celery/backends/mongodb.py

```diff
@@ -141,8 +141,7 @@ def _get_connection(self):
                 # This enables the use of replica sets and sharding.
                 # See pymongo.Connection() for more info.
                 host = self.host
-                if isinstance(host, string_t) \
-                   and not host.startswith('mongodb://'):
+                if isinstance(host, string_t) and '://' not in host:
```

Comment:
An alternative suggestion:

```
if isinstance(host, string_t) and not host.replace('+srv', '').startswith('mongodb://'):
```

The above will help identify directly the use case in the future. What do you think?

Reply:
No, that's wrong. The MongoClient does all the DNS lookups. We don't want to mess with a mongodb+srv url because we'll do it wrong. See the test for what I mean.

Reply:
I think that @josephtate point is correct, you want to consider `host` the host if it doesn't include any schema, and a schema is always specified through the usage of `://`. So it's fairly more reliable to check for the schema separator than checking for `mongodb://` because if mongo introduces new kind of schemas, celery we will continue to support them this was.

Reply:
I believe this sounds as the more correct approach, thanks for clarifying.